streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java

```diff
@@ -1570,4 +1611,30 @@ private void addRecord(final MockConsumer<byte[], byte[]> mockConsumer,
             new byte[0],
             new byte[0]));
     }
+
+
+    /**
+     * Mock consumer that could alter stream thread state during #poll()
+     */
+    private abstract class MockStreamThreadConsumer<K, V> extends MockConsumer<K, V> {
+
+        private StreamThread streamThread;
+
+        MockStreamThreadConsumer(final OffsetResetStrategy offsetResetStrategy) {
+            super(offsetResetStrategy);
+        }
+
+        @Override
+        public synchronized ConsumerRecords<K, V> poll(final Duration timeout) {
+            assertNotNull(streamThread);
+            onPoll();
+            return super.poll(timeout);
+        }
+
+        public abstract void onPoll();
```

Comment:
Why do we need this method? Can we call `streamThread.shutdown();` directly in `poll(...)` ?

Reply:
Just want to have this mock consumer being reused later if someone needs to inject different behavior